### DISCUSSION TITLE
Improve gradle build script

### DISCUSSION THREAD

**Discussion by toliner** [dipien/releases-hub-gradle-plugin#149] [General]
First of all, I would like to thank you for making this project.This is exactly what I was looking for.
I've tried to make something similar to this, but I've failed many times.

At first, I read `*.gradle` files of this project and found the following problems:

- However the main code is Kotlin, build the build scripts are not Kotlin but Groovy.
- old style of applying plugin (`apply plugin: "id"` style) is used.
- deprecated dependency declaration `compile` is used.
- duplicated code of `repositories{}` block.

I thing Gradle Kotlin DSL is much better than Groovy, but there are still some problems, so I'd like to hear your thoughts on that.

The use of `plugins{}` block to apply plugin and `implementation` dependency declaration is strongly recommended, so these should be fixed. 

I have nothing to say about code duplication. This may solved by moving to Gradle Kotlin DSL, because you can use  extension function to `RepositoryHandler`.

P.S. I'm not good at writing English and use machine translation in several places.

**Comment by maxirosson** [comment]
Thanks for your feedback. I created some issues:
- Migrate all build.gradle to Kotlin DSL https://github.com/releaseshub/releases-hub-gradle-plugin/issues/74
- Support plugins DSL https://github.com/releaseshub/releases-hub-gradle-plugin/issues/56

I already switched from `compile` to `implementation`
